test/pkg/environment/common/expectations.go

```diff
@@ -261,11 +261,12 @@ func (env *Environment) ExpectPrefixDelegationDisabled() {
 		"ENABLE_PREFIX_DELEGATION", "false", "aws-node")
 }
 
-func (env *Environment) ExpectExists(obj client.Object) {
+func (env *Environment) ExpectExists(obj client.Object) client.Object {
```

Comment:
I think using it as a client.Object is a good generic use here. I guess if you use a generic type, you just move the cast into the function itself, but then you lose the compile time validation that the object you're passing in is actually castable into a client.Object, which I think is more important here, rather than forcing users to do ExpectExists() and cast it with the object type that they're expecting.